share/toolchain-nxdk.cmake

```diff
@@ -45,4 +52,34 @@ set(_CMAKE_C_IPO_SUPPORTED_BY_CMAKE YES)
 set(_CMAKE_C_IPO_MAY_BE_SUPPORTED_BY_COMPILER YES)
 set(CMAKE_C_COMPILE_OPTIONS_IPO -flto)
 
+# TODO: CMAKE_MODULE_PATH is intended for project level than toolchain,
+#       find out what's wrong with CMAKE_FIND_ROOT_PATH as it is not working for some reason?
+set(CMAKE_MODULE_PATH "${NXDK_DIR}/share/cmake/Modules")
 set(PKG_CONFIG_EXECUTABLE "${NXDK_DIR}/bin/nxdk-pkg-config" CACHE STRING "Path to pkg-config")
+
+# Additional toolchain steps required for create xbe and xiso files.
+macro(add_executable)
+  # Perform normal call since there's no modification needed except for additional steps require
+  _add_executable(${ARGV})
+
+  # If author request generate iso file, then prepare the command here.
+  if(GEN_XISO)
+    set(XISO_OUTPUT "${NXDK_DIR}/tools/extract-xiso/build/extract-xiso" -c "$<TARGET_FILE_DIR:${ARGV0}>/bin" "$<TARGET_FILE_DIR:${ARGV0}>/${GEN_XISO}")
+  endif()
```

Comment:
I suggest to leave XISO out of this. It also doesn't feel CMake-ish.

It also doesn't handle a common use-case where a user will want to selectively add more files to the XISO; so this will need a better design.

Also, if we support `GEN_XISO` now, then we can't remove it without breaking projects.
As nxdk matures, we should avoid breakage like that.

---

[In my own CMake work I just created another CMake function](https://github.com/JayFoxRox/nxdk/pull/74/files#diff-f4116787576a8fd4b6cba2c395d8f2fa515c539bec4c101de9cf47bc77d3dce4R33-R38) but I believe I still did XISO creation manually in a shell script for flexibility.

Reply:
I like the concept of making it as CMake function which I now notice from you. Initially I'm following how makefile handling things which currently use `GEN_XISO`. If preferred the change to CMake function, I'll be happily change it. Yet may need to preserve backward compatibility for the change from make to CMake?

Reply:
> Yet may need to preserve backward compatibility for the change from make to CMake?

I don't think so.

Users who move from the makefile to CMake will have to change their code anyway, so they can add the function call themselves.

I believe the original nxdk makefile (for use outside of nxdk internally) will also be retired eventually, and users who still want to use a makefile can use their own OR find some build-system they like (which doesn't have to be CMake).
Also [see my work on autoconf etc.](https://github.com/JayFoxRox/nxdk/pull/83/files) - while CMake support is part of nxdk, it's trivial to use other build-systems after the existing refactors to the build-system (which only added CMake, for better or worse)

Reply:
Is there opened ticket discussed about it and/or have list of tasks for moving away from original nxdk makefile to have it done through toolchain script? I am currently unable to find the information in the repository. If there is or isn't, it will be helpful to follow through guideline to make a working toolchain as possible and have tasks done.

Reply:
> Xiso creation is one of those - it gives very little control over the end result (on-disc layout, non-1:1 directory mapping etc.), and technically isn't required to get something running on the Xbox, so imho we can leave that out the CMake toolchain file. It's still frequently used for quick emulator testing, so my suggestion would be to have a wiki entry "Creating an xiso from CMake" or something that supplies code that can just be dropped in.

[Quote from thrimbor](https://github.com/XboxDev/nxdk/pull/631#issuecomment-1646725220)

In summary, it's better off to perform xiso creation from application preferred target's POST_BUILD process with CMake's add_custom_command function; via nxdk wiki CMake's getting started/howto.